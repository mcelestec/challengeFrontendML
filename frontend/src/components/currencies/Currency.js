import React from 'react'
import PropTypes from 'prop-types';

const Currency = ({ currency, amount, decimals }) => {
    return (
        <div className="currency__general">
            <span className="currency__general-amount" data-testid="currency-amount">{(amount).toLocaleString("es-ar",
                {
                    style: "currency",
                    currency: `${currency}`,
                    maximumFractionDigits: 0
                })}
            </span>
            {(decimals && decimals > 0) && <span className="currency__general-decimals" data-testid="currency-decimals">{decimals.toString().padStart(2, 0)}</span>}
        </div>
    )
}

Currency.propTypes = {
    currency: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    decimals: PropTypes.number
}


export default Currency