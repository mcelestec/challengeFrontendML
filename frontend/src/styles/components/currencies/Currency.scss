.currency__general {
    display: inline-block;
}

.currency__general-decimals {
    display: inline-block;
    transform: scale(0.7) translate(-10%, -10%);
}

.currency__general-amounts {
    display: inline-block;
}