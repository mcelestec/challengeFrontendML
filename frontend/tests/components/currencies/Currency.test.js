import { render, screen } from "@testing-library/react";
import Currency from "../../../src/components/currencies/Currency";

describe('Test: probando <Currency />', () => {

    const currency_ars = 'ARS';
    const currency_usd = 'USD';
    const decimal_0 = 0;
    const decimal_less_10 = 7;
    const decimal_more_10 = 50;
    const amount_less_1000 = 900;
    const amount_more_1000 = 100800;

    test('Llamada a Currency: debe hacer match con el snapshot', () => {
        const { container } = render(<Currency currency={currency_ars} amount={amount_less_1000} decimals={amount_more_1000} />);
        expect(container).toMatchSnapshot();
    }
    );

    test('Llamada a Currency: validar props obligatorios', () => {

        render(<Currency currency={currency_ars} amount={amount_less_1000} decimals={decimal_more_10} />);

    }
    );

    test('Llamada a Currency: 900.50 pesos argentinos', () => {

        render(<Currency currency={currency_ars} amount={amount_less_1000} decimals={decimal_more_10} />);

        expect(screen.getByTestId('currency-amount').innerHTML).toBe(`$&nbsp;900`);
        expect(screen.getByTestId('currency-decimals').innerHTML).toBe(`${decimal_more_10}`);
    }
    );

    test('Llamada a Currency: 100800 pesos argentinos (decimales 0)', () => {

        render(<Currency currency={currency_ars} amount={amount_more_1000} decimals={decimal_0} />);

        expect(screen.getByTestId('currency-amount').innerHTML).toBe(`$&nbsp;100.800`);
        expect(screen.queryByTestId('currency-decimals')).toBeNull();
    }
    );

    test('Llamada a Currency: 900.50 dólares', () => {

        render(<Currency currency={currency_usd} amount={amount_less_1000} decimals={decimal_more_10} />);

        expect(screen.getByTestId('currency-amount').innerHTML).toBe(`US$&nbsp;900`);
        expect(screen.getByTestId('currency-decimals').innerHTML).toBe(`${decimal_more_10}`);
    }
    );

    test('Llamada a Currency: 100800 dólares (decimales 0)', () => {

        render(<Currency currency={currency_usd} amount={amount_more_1000} decimals={decimal_0} />);

        expect(screen.getByTestId('currency-amount').innerHTML).toBe(`US$&nbsp;100.800`);
        expect(screen.queryByTestId('currency-decimals')).toBeNull();
    }
    );

    test('Llamada a Currency: 100800 dólares (sin decimales)', () => {

        render(<Currency currency={currency_usd} amount={amount_more_1000} />);

        expect(screen.getByTestId('currency-amount').innerHTML).toBe(`US$&nbsp;100.800`);
        expect(screen.queryByTestId('currency-decimals')).toBeNull();
    }
    );

    test('Llamada a Currency: decimales mayores a 0 y menores a 10', () => {

        render(<Currency currency={currency_usd} amount={amount_less_1000} decimals={decimal_less_10} />);

        expect(screen.getByTestId('currency-decimals').innerHTML).toBe(`0${decimal_less_10}`);
    }
    );
});