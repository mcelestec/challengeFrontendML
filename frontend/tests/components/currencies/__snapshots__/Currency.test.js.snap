// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: probando <Currency /> Llamada a Currency: debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="currency__general"
  >
    <span
      class="currency__general-amount"
      data-testid="currency-amount"
    >
      $Â 900
    </span>
    <span
      class="currency__general-decimals"
      data-testid="currency-decimals"
    >
      100800
    </span>
  </div>
</div>
`;
